app/src/main/java/com/randomappsinc/studentpicker/home/DeleteListDialog.java

```diff
@@ -13,20 +13,22 @@
         void onDeleteListConfirmed(int position);
     }
 
-    private MaterialDialog dialog;
+    public MaterialDialog dialog;
     private int position;
 
     DeleteListDialog(@NonNull Listener listener, Context context) {
         dialog = new MaterialDialog.Builder(context)
                 .title(R.string.confirm_deletion_title)
-                .content(R.string.confirm_deletion_message)
                 .positiveText(android.R.string.yes)
                 .negativeText(android.R.string.cancel)
                 .onPositive((dialog, which) ->
                         listener.onDeleteListConfirmed(position)
                 ).build();
     }
 
+    // Instead of sending the content inside the show function, Defining the dialog object as public is better to get an access for it from the MainActivity
```

Comment:
Hmm, this is a good point that setting the content within this method violates single responsibility design. However, exposing the dialog breaks encapsulation since anyone can now mutate the dialog and this mutation is now increasing the complexity of the activity, which we want to avoid. What if we made the method presentForList(int position, String listName)?

Reply:
![Screenshot_20191221-105447](https://user-images.githubusercontent.com/32240376/71305704-660d1000-23e0-11ea-981f-555fa5d68c24.png)
![Screenshot_20191221-105450](https://user-images.githubusercontent.com/32240376/71305705-660d1000-23e0-11ea-98c6-d2ad28bb348c.png)

Reply:
Thanks for the feedback, but I can't see a function called presentForList(int position, String listName)?
I created a setter ruction inside the DeleteListDialog and defined MaterialDialog as private again.